### PR TITLE
Add syncer for retrieving account balance

### LINKED ISSUE #10: BTC Syncer
Implement this code [BTC Testnet](https://github.com/herdius/herdius-core/blob/master/syncer/btctestnet.go) in Validator.
Need to make it generic so that it will work for both main network and test network.

### PR DESCRIPTION
**Pull request by cuonglm**
Fixes #10

### CODE REVIEW

File: validator/service/syncer_test.go [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestNewSyncer(t *testing.T) {
+	b, err := NewSyncer("btc", "test3", "d3b5ec2c94cc4ba183f02aca9d8729da")
```
Comment (lmahanand) [inline validator/service/syncer_test.go:11]: We should read the api keys from environment variables or AWS secret manager.

File: validator/service/syncer.go [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+	coin  string
+	token string
+
+	api blockcypher.API
+}
+
+var _ Syncer = (*blockCypher)(nil)
+
+func (b *blockCypher) Balance(address string) (int, error) {
+	addr, err := b.api.GetAddrFull(address, nil)
```
Comment (lmahanand) [inline validator/service/syncer.go:31]: Could you please write a test case if it works for lite coin as well?

**Review by lmahanand — CHANGES_REQUESTED** [review]

File: validator/service/syncer_test.go (hunk 1 above)
Comment (cuonglm) [reply validator/service/syncer_test.go:11]: This is part of the init config, e.g:

```
token := config.GetToken()
b, err := NewSyncer("btc", "test3", token)
```

Comment (lmahanand) [reply validator/service/syncer_test.go:11]: Actually the reason is what the value ** d3b5ec2c94cc4ba183f02aca9d8729da** is a API key. Keeping it in an open source code will not be good.

Comment (cuonglm) [reply validator/service/syncer_test.go:11]: Ah right, i though this is test key only (I copied it from herdius-core).

File: validator/service/syncer.go (hunk 2 above)
Comment (cuonglm) [reply validator/service/syncer.go:31]: Added

File: validator/service/syncer_test.go (hunk 1 above)
Comment (cuonglm) [reply validator/service/syncer_test.go]: Fixed

**Review by lmahanand — APPROVED** [review]

**Review by balazsdeme — APPROVED** [review]

**Pull request merged**